### PR TITLE
CNF definition

### PR DESCRIPTION
**Pull request by jeffsaelens**
Update to the glossary with second crack at the CNF definition. The term cloud native will be referenced with the assumption that it too will be spelled out in more detail in the glossary. This PR will be for a single definition that the group can focus on. Please reference discussions #136 and #27 .

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ For the purpose of this document, “Company” shall mean any entity (Company-A
 Each organization listed in the [interested parties document](interested-parties.md) has one vote. Interested parties can be added at any time, but must be added at least one week before any election to have a vote. Any contributor from an organization may cast the vote for that organization. Each organization can cast one vote for a co-chair candidate in each of the communities (Kubernetes (K8s) Community, Service Provider (SP) Community, and CNF Developer (Dev) Community), three total votes.
 
 ### Voting Procedure
+
```
Comment (taylor) [inline GOVERNANCE.md:62]: note: this is to fix the linter complaints

File: doc/glossary.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 **Cloud Native** - TBD
 
-**Cloud Native Network Function** - TBD
+**Cloud-Native Network Function** - A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF consists of one or more microservices and has been developed using Cloud Native Principles including immutable infrastructure, declarative APIs, and a “repeatable deployment process”.
```
Comment (taylor) [inline doc/glossary.md]:
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF consists of one or more microservices which can be delivered and has been developed using Cloud Native Principles including loose coupling, immutable infrastructure, declarative APIs, and a “repeatable deployment process”.
```

Comment (taylor) [reply doc/glossary.md]: @jeffsaelens added delivery since we talk about both development and the deployment of CNFs.

**Review by taylor — CHANGES_REQUESTED** [review]

File: doc/glossary.md (hunk 2 above)
Comment (jeffsaelens) [reply doc/glossary.md]: @xmulligan pulled the hyphens out of the other pages. Turns out the style guide tosses grammar out with the bath water ;-) other than that, I like the changes.

**Review by CsatariGergely — APPROVED** [review]

**Review by xmulligan — APPROVED** [review]

**Review by taylor — APPROVED** [review]
lgtm

**Comment by electrocucaracha** [comment]
I'm okay with this definition but IMHO this relies on the Cloud-Native application definition which is in the same document and it's not defined yet so it seems to partially defining it.

**Comment by xmulligan** [comment]
@electrocucaracha that's actually a really good point, we are defining it with something we haven't defined and isn't really defined elsewhere

**Comment by jeffsaelens** [comment]
> @electrocucaracha that's actually a really good point, we are defining it with something we haven't defined and isn't really defined elsewhere

Agreed, why I called out in the PR that this definition is assuming that we are going to more clearly define Cloud Native above in this glossary. Its the loose definition of that term that makes everything else more difficult to define.

**Comment by electrocucaracha** [comment]
> > @electrocucaracha that's actually a really good point, we are defining it with something we haven't defined and isn't really defined elsewhere
> 
> Agreed, why I called out in the PR that this definition is assuming that we are going to more clearly define Cloud Native above in this glossary. Its the loose definition of that term that makes everything else more difficult to define.

Maybe we can use/refer to the [CNCF glossary definition](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md)

**Comment by sishbi** [comment]
I approve...

File: doc/glossary.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 **Cloud Native** - TBD
 
-**Cloud Native Network Function** - TBD
+**Cloud Native Network Function** - A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF consists of one or more microservices which can be delivered and has been developed using Cloud Native Principles including loose coupling, immutable infrastructure, declarative APIs, and a “repeatable deployment process”.
```
Comment (jeffsaelens) [inline doc/glossary.md]:
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF has been developed using Cloud Native Principles and allows for a “repeatable deployment process”.
```

**Review by jeffsaelens — COMMENTED** [review]
Check newest iteration

**Review by sishbi — APPROVED** [review]

**Review by Smitholi67 — APPROVED** [review]

File: doc/glossary.md (hunk 3 above)
Comment (electrocucaracha) [reply doc/glossary.md]: What about adding a link to the glossary definition 
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) that implements network functionality. A CNF has been developed using Cloud Native Principles and allows for a “repeatable deployment process”.
```

File: doc/glossary.md [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 **Cloud Native** - TBD
 
-**Cloud Native Network Function** - TBD
+**Cloud Native Network Function** - A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF has been developed using Cloud Native Principles and allows for a “repeatable deployment process”.
```
Comment (taylor) [inline doc/glossary.md]:
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) that implements network functionality. A CNF has been developed using Cloud Native Principles and allows for a “repeatable deployment process”.
```

**Review by electrocucaracha — APPROVED** [review]

File: doc/glossary.md [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 **Cloud Native** - TBD
 
-**Cloud Native Network Function** - TBD
+**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) that implements network functionality. A CNF has been developed using Cloud Native Principles and allows for a “repeatable deployment process”.
```
Comment (taylor) [inline doc/glossary.md]:
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) that implements network functionality. A CNF is developed using Cloud Native Principles to enable desired behavior such as “repeatable deployment process”.
```

Comment (taylor) [reply doc/glossary.md]: Repeatable deployment process is a desired behaviour.

Comment (taylor) [inline doc/glossary.md]:
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) developed using Cloud Native principles that provides network functionality. 
```

File: doc/glossary.md [hunk 6]
```diff
@@ -6,7 +6,7 @@
 
 **Cloud Native** - TBD
 
-**Cloud Native Network Function** - TBD
+**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) that implements network functionality. A CNF is developed using Cloud Native Principles to enable desired behavior such as “repeatable deployment process”.
```
Comment (taylor) [inline doc/glossary.md]:
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) developed using Cloud Native principles that provides network functionality. 
```

- Pankaj

**Review by v1k0d3n — APPROVED** [review]
LGTM

**Review by electrocucaracha — APPROVED** [review]

File: doc/glossary.md [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
 **Cloud Native** - TBD
 
-**Cloud Native Network Function** - TBD
+**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) developed using Cloud Native principles that provides network functionality.
```
Comment (taylor) [inline doc/glossary.md]:
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md), developed using cloud native principles, that provides network functionality. 
```

**Review by v1k0d3n — APPROVED** [review]
LGTM

File: doc/glossary.md [hunk 8]
```diff
@@ -6,7 +6,7 @@
 
 **Cloud Native** - TBD
 
-**Cloud Native Network Function** - TBD
+**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md), developed using cloud native principles, that provides network functionality.
```
Comment (taylor) [inline doc/glossary.md]:
```suggestion
**Cloud Native Network Function** - A cloud native network function (CNF) is a [cloud native application](https://github.com/cncf/glossary/blob/main/definitions/cloud_native_apps.md) that provides network functionality, which is developed using cloud native principles. 
```

**Pull request merged**